### PR TITLE
fix/caching-pubkey-mapping

### PR DESCRIPTION
**Pull request by siepra**

### CODE REVIEW

File: packages/mobile/src/store/init/startConnection/startConnection.saga.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export function* startConnectionSaga(
 }
 
 function* setConnectedSaga(socket: Socket): Generator {
-  // @ts-expect-error
+  /// @ts-expect-error
```
Comment (vinkabuki) [inline packages/mobile/src/store/init/startConnection/startConnection.saga.ts:22]: ?

**Review by vinkabuki — APPROVED** [review]

File: packages/mobile/src/components/Chat/Chat.test.tsx [hunk 2]
```diff
@@ -3903,7 +3903,7 @@ describe('Chat component', () => {
                     </View>
                   </View>
                 </View>
-              </View>
+              </div>
```
Comment (kowalski) [inline packages/mobile/src/components/Chat/Chat.test.tsx:3906]: this look suspicious

there are no divs in mobile. What is this doing here?

**Review by kowalski — APPROVED** [review]
Looks good, safe for this weird `</div>`

File: packages/mobile/src/components/Chat/Chat.test.tsx (hunk 2 above)
Comment (siepra) [reply packages/mobile/src/components/Chat/Chat.test.tsx:3906]: That's super weird. It was supposed to be the opposite. Must have done something wrong when merging master

File: packages/nectar/src/sagas/messages/verifyMessage/verifyMessages.saga.ts [hunk 3]
```diff
@@ -25,9 +25,7 @@ function* verifyMessage(message: ChannelMessage, crypto: SubtleCrypto): Generato
   const publicKeysMapping = yield* select(messagesSelectors.publicKeysMapping)
 
   let cryptoKey = publicKeysMapping[message.pubKey]
-  /* Sometimes an empty object is being stored as a key (async call problem?)
-     checking its content (key type) prevents invalid object from being passed to verification method */
-  if (!cryptoKey || !cryptoKey.type) {
+  if (cryptoKey === undefined) {
```
Comment (holmesworcester) [inline packages/nectar/src/sagas/messages/verifyMessage/verifyMessages.saga.ts:28]: Are there any cases where it isn't undefined but it's invalid in some other way? What will happen if so?

**Review by holmesworcester — APPROVED** [review]

File: packages/nectar/src/sagas/messages/verifyMessage/verifyMessages.saga.ts (hunk 3 above)
Comment (siepra) [reply packages/nectar/src/sagas/messages/verifyMessage/verifyMessages.saga.ts:28]: That was the original source of the problem. Redux persist store was not able to correctly serialize/deserialize CryptoKey mapping which resulted in invalid data being present in the store - but it was incorrect in a different ways depending on the platform (web, mobile). That's why I added custom transformations for this type of data which simply ignores this mapping when flushing store state by setting it as undefined

**Pull request merged**